### PR TITLE
Bugfixes 18X, lerna, remove codependencies, update express

### PR DESCRIPTION
**Pull request by kamilmysliwiec**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+
+## 4.1.4
+- **common**: remove deprecated `@ExceptionFilters()` decorator
+- **core**: update to latest **express* version (4.16.2)
+- **core**: bugfix #187
+- **core**: bugfix #185
+- **all**: packages codependency refactor
+- **all**: add *lerna* package
```
Comment (wbhob) [inline CHANGELOG.md]: yay!

File: gulpfile.js [hunk 2]
```diff
@@ -26,3 +28,8 @@ modules.forEach((module) => {
     });
 });
 
+gulp.task('build', function (cb) {
```
Comment (wbhob) [inline gulpfile.js]: yay! good to finally have this functionality.

File: lerna.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "lerna": "2.4.0",
+  "packages": [
+    "lib/*"
+  ],
+  "version": "4.1.7"
```
Comment (wbhob) [inline lerna.json]: should be 4.1.4

File: src/core/nest-factory.ts [hunk 4]
```diff
@@ -6,12 +7,14 @@ import { NestModuleMetatype } from '@nestjs/common/interfaces/modules/module-met
 import { Logger } from '@nestjs/common/services/logger.service';
 import { messages } from './constants';
 import { NestApplication } from './nest-application';
-import { NestMicroservice } from './nest-microservice';
 import { isFunction } from '@nestjs/common/utils/shared.utils';
-import { MicroserviceConfiguration } from '@nestjs/microservices/interfaces/microservice-configuration.interface';
+import { MicroserviceConfiguration } from '@nestjs/common/interfaces/microservices/microservice-configuration.interface';
```
Comment (wbhob) [inline src/core/nest-factory.ts]: Why not merge common into core? It would clean up a lot of confusion.

File: src/core/nest-factory.ts [hunk 5]
```diff
@@ -46,9 +49,13 @@ export class NestFactoryStatic {
         module,
         config?: MicroserviceConfiguration): Promise<INestMicroservice> {
 
+        if (!NestMicroservice) {
+          throw new MicroservicesPackageNotFoundException();
```
Comment (wbhob) [inline src/core/nest-factory.ts]: great workaround

**Comment by wbhob** [comment]
Also @kamilmysliwiec, just as an aside, I think the `help-wanted` tag on issues and PRs is typically used to indicate where you are requesting commits/reviews, not the openner needing help with an issue. As an alternative, I would love for there to be an `RFC` tag - Request for Commits

**Comment by kamilmysliwiec** [comment]
@wbhob, gotcha, my mistake 😺 thanks

**Pull request merged**